webui/settings.py

```diff
@@ -247,7 +247,7 @@ def get_namespaces_for(feature: str) -> list:
     @staticmethod
     def get_providers_for(feature: str, namespace: str):
         return ConfigFile().getList('{}.namespace.{}/providers'.format(feature, namespace),
-                                    ['ec2', 'azure', 'gce'])
+                                    ConfigFile().getList('{}/providers'.format(feature), ['ec2', 'azure', 'gce']))
```

Comment:
This seems like it should be just `'{}/providers'.format(feature)` as parameter inside the `ConfigFile().getList` but you call it again the `getList`. i havent checked how the implementation is for `getList` but if this is above is indented maybe the getList could be refactored.

Reply:
yeah looks like you not getting the purpose . please read commit message of this PR and implementation of getList . in case you will still concerns after this I can try to explain ...

Reply:
i think i have. but to be honest i found it confusing. it is kinda of if...elif..else but kinda hidden. Anyway i am taking back the request for change but i have one last observation

if i get it right this line                     
`raise LookupError('Missing attribute {} in file {}'.format(config_path, self.filename))` 
is never evaluated. is that ok? Do we care if this list will not found in the config?

Reply:
no we don't , in such case we return **all** providers . exception which you mention is for more generic cases